### PR TITLE
ci: Add PR template, delete codeowners

### PR DESCRIPTION
**Pull request by AntonLydike**
This PR adds a PR template with a checklist on what to do before submitting a PR.

@math-fehr @webmiche @superlopuh

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+---
+
+Thank you for submitting a PR to xDSL!
+
+Make sure to follow the steps in the checklist below *before* you submit this PR!
+
+## Checklist before you submit this PR:
+
+ - [ ] If your PR is not ready for review, open it as a draft instead.
+ - [ ] Make sure your PR title is [following the guidelines](https://github.com/xdslproject/xdsl/wiki/Tags-for-commit-messages-and-PR-titles)
```
Comment (superlopuh) [inline .github/pull_request_template.md:12]:
```suggestion
 - [ ] Make sure your PR title follows the [guidelines](https://github.com/xdslproject/xdsl/wiki/Tags-for-commit-messages-and-PR-titles)
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ - [ ] Add some tags to categorize the PR
+ - [ ] Request review from people you *know* have to sign off on these changes, you can leave it blank for now if you are not sure.
+ - [ ] Tag people you wish to get feedback from (using @\<username>), they will then sort out the reviewers.
```
Comment (superlopuh) [inline .github/pull_request_template.md:15]:
```suggestion
 - [ ] Tag people you wish to get feedback from (using @\<username>), they will then sort out the reviewers.
 - [ ] Assign this PR to yourself
```

**Review by superlopuh — APPROVED** [review]
ty

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+Thank you for submitting a PR to xDSL!
+
+Make sure to follow the steps in the checklist below *before* you submit this PR!
+
+## Checklist before you submit this PR:
+
+ - [ ] If your PR is not ready for review, open it as a draft instead.
+ - [ ] Make sure your PR title follows the [guidelines](https://github.com/xdslproject/xdsl/wiki/Tags-for-commit-messages-and-PR-titles)
+ - [ ] Add some tags to categorize the PR
```
Comment (math-fehr) [inline .github/pull_request_template.md:13]: This does not work for new users, should we change the text to reflect that?

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ - [ ] Request review from people you *know* have to sign off on these changes, you can leave it blank for now if you are not sure.
```
Comment (math-fehr) [inline .github/pull_request_template.md:14]: Can new users request reviewers? I guess that they don't need to and we'll do for them in that case. In any way, a new users wouldn't know who is responsible for what.

**Review by math-fehr — APPROVED** [review]
Nice!

**Review by webmiche — APPROVED** [review]
I am not sure whether we want to remove the codeowners stuff already. I agree with @superlopuh that we could probably make this work properly in order to give a list of people who *must* approve a PR or something.

But we can also just add it again later on, both works for me

File: .github/pull_request_template.md (hunk 4 above)
Comment (georgebisbas) [reply .github/pull_request_template.md:14]: Note: Pull request authors can't request reviews unless they are either a repository owner or collaborator with write access to the repository.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review

File: .github/pull_request_template.md (hunk 3 above)
Comment (georgebisbas) [reply .github/pull_request_template.md:13]: I think as soon as their title is fine, the person who will click the button to merge should be the one to hold responsibility for the label/tag being there

**Comment by kingiler** [comment]
I would suggest pin the PR template (as well as issue template) so people will notice them instead of finding them at wiki.

**Comment by webmiche** [comment]
This should have integration with github, so it should be displayed when someone tries to open a PR.

**Pull request merged**